blocks/search.py

```diff
@@ -277,6 +276,7 @@ def search(self, input_values, eol_symbol, max_length,
             negative log-likelihood.
 
         """
+        self.beam_size = input_values[input_values.keys()[0]].shape[1]
```

Comment:
I am afraid this is not a generic solution. A sequence generator can simply have no sequential inputs, consider an encoder-decoder model.

However, after #719 is solved the beam size seems to be not necessary at all!